src/modules/esl-event-listener/core/targets/intersection.event.ts

```diff
@@ -22,14 +27,25 @@ export class ESLIntersectionEvent extends Event implements IntersectionObserverE
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/IntersectionObserverEntry/time) */
   public readonly time: DOMHighResTimeStamp;
 
-  protected constructor(target: Element) {
-    super(ESLIntersectionEvent.type, {bubbles: false, cancelable: false});
+  protected constructor(target: Element, eventName: typeof ESLIntersectionEvent.type) {
+    super(eventName, {bubbles: false, cancelable: false});
```

Comment:
The order of the original constructor is preferable when it makes sense.
```suggestion
  protected constructor(eventName: typeof ESLIntersectionEventType, target: Element) {
    super(eventName, {bubbles: false, cancelable: false});
```